pandas/core/series.py

```diff
@@ -1230,6 +1230,13 @@ def setitem(key, value):
                 if _is_unorderable_exception(e):
                     raise IndexError(key)
 
+                if is_scalar(key) and not is_integer(key) and key not in self.index:
```

Comment:
i would trace further

you are adding a new case whereby it’s likely an existing path is just incorrect somewhere

Reply:
the path here is above when we call `self._set_with_engine(key, value)`, `_set_with_engine` calls `self.index._engine.set_value(self._values, key, value)`.  The engine expects `self._values` to be an ndarray, but here it is a `DatetimeArray`.  Eventually it _might_ be worth addressing this at the low-level engine level, but definitely out of scope for now.

Reply:
FWIW, that's where I got stuck trying to fix this same bug. I tried converting it to an ndarray before passing it into the engine but a lot of other routines failed.

Reply:
move this to _set_with and its better (after the cases are narrowed down)

Reply:
sure